src/ethereum_test_specs/helpers.py

```diff
@@ -1,71 +1,93 @@
 """Helper functions."""
 
 from dataclasses import dataclass
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Literal
 
 import pytest
 
 from ethereum_clis import Result
-from ethereum_test_exceptions import ExceptionBase, ExceptionMapper, UndefinedException
+from ethereum_test_exceptions import (
+    BlockException,
+    ExceptionBase,
+    ExceptionWithMessage,
+    TransactionException,
+    UndefinedException,
+)
 from ethereum_test_types import Transaction, TransactionReceipt
 
 
-class TransactionUnexpectedSuccessError(Exception):
+class UnexpectedSuccessError(Exception):
     """Exception used when the transaction expected to fail succeeded instead."""
 
-    def __init__(self, index: int, nonce: int):
-        """Initialize the exception with the transaction index and nonce."""
+    def __init__(self, ty: Literal["Block", "Transaction"], **kwargs):
```

Comment:
`ty` is a little too skimpy for my liking, how do you like `execution_context`? I also made a suggestion for this in #1404.